### PR TITLE
Add dispatch abstraction for admintools -t list_allnodes

### PR DESCRIPTION
**Pull request by spilchen**
This is a continuation of the change to hide all admintools commands behind an abstraction so that we can replace them with the vclusterOps library. This adds an abstraction for the call to admintools -t list_allnodes. I called it FetchNodeState.

### CODE REVIEW

File: pkg/vadmin/fetch_node_state_at.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		return nil, res, err2
+	}
+	return a.parseClusterNodeStatus(stdout, a.buildHostsNeededSet(s.Hosts)), ctrl.Result{}, nil
+}
+
+// parseClusterNodeStatus will parse the output from a AT -t list_allnodes call
+func (a Admintools) parseClusterNodeStatus(stdout string, hostsNeeded map[string]bool) map[string]string {
+	stateMap := map[string]string{}
+	lines := strings.Split(stdout, "\n")
+	const ColHeaderCount = 2
```
Comment (roypaulin) [inline pkg/vadmin/fetch_node_state_at.go:48]: If I understand this variable represent the first 2 lines (the header). I think the name is a bit confusing. Would something like `headerRowCount` or `headerRowSize` be better?

File: pkg/vadmin/fetch_node_state_at.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	"strings"
+
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	"github.com/vertica/vertica-kubernetes/pkg/names"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/fetchnodestate"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// FetchNodeState will determine if the given set of nodes are considered UP
+// or DOWN in our consensous state. It returns a map of vnode to its node state.
```
Comment (roypaulin) [inline pkg/vadmin/fetch_node_state_at.go:29]:
```suggestion
// or DOWN in our consensus state. It returns a map of vnode and its node state.
```
Do you mean consensus?

File: pkg/vadmin/fetch_node_state_at_test.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+				" v_d_node0002 | 10.244.1.7 | DOWN  | vertica-11.0.0.20210309 | d \n" +
+				" v_d_node0003 | 10.244.1.8 | UP    | vertica-11.0.0.20210309 | d \n" +
+				"\n",
+			},
+		}
+		evWriter := mgmterrors.TestEVWriter{}
+		d := MakeAdmintools(logger, vdb, fpr, &evWriter)
+		state, res, err := d.FetchNodeState(ctx,
+			fetchnodestate.WithInitiator(atPod, "10.244.1.6"),
+			fetchnodestate.WithHost("v_d_node0001", "10.244.1.6"),
```
Comment (roypaulin) [inline pkg/vadmin/fetch_node_state_at_test.go:51]: What happens when you pass the same host twice?

**Review by roypaulin — COMMENTED** [review]
Looks good!

File: pkg/vadmin/fetch_node_state_at.go (hunk 1 above)
Comment (spilchen) [reply pkg/vadmin/fetch_node_state_at.go:48]: I'll change it to `headerRowCount`

**Review by roypaulin — APPROVED** [review]

**Pull request merged**